src/NServiceBus.Core/Routing/RoutingFeature.cs

```diff
@@ -87,5 +96,35 @@ protected internal override void Setup(FeatureConfigurationContext context)
                 }
             }
         }
+
+        void EnableBestPracticeEnforcement(FeatureConfigurationContext context)
+        {
+            var validations = new Validations(context.Settings.Get<Conventions>());
+
+            context.Pipeline.Register(
+                "EnforceSendBestPractices",
+                new EnforceSendBestPracticesBehavior(validations),
+                "Enforces send messaging best practices");
+
+            context.Pipeline.Register(
+                "EnforceReplyBestPractices",
+                new EnforceReplyBestPracticesBehavior(validations),
+                "Enforces reply messaging best practices");
+
+            context.Pipeline.Register(
```

Comment:
Yes but wouldn't routing be invoked when you call .Subscribe and throw even if this wasn't there?